### PR TITLE
Handle permission errors of urls.txt reading

### PR DESCRIPTION
**Pull request by bfis**
The `UrlsData` class' `__init__` attempts to read the corresponding `urls.txt` while silently ignoring it's absence. In particular this (currently) does include insufficient permission do interact with the/any containing directory, but not the read permission of the file itself. This PR fixes this.

Example of a `pkgs_dir` causing (unexpected) issues (for "others"), since `isfile("pkgs_dir/urls.txt") is True` but the file (and entire `pkgs_dir`) is not readable/usable:

```
drwxrwx--x pkgs_dir/
-rw-rw---- pkgs_dir/urls.txt
```

┆Issue is synchronized with this [Jira Task](https://anaconda.atlassian.net/browse/CONDAORG-94)

### CODE REVIEW

**Comment by mcg1969** [comment]
This is definitely fine, just waiting on the CLA

**Comment by mcg1969** [comment]
Looks like this isn't Python 2 compatible, and we're not quite ready to drop that... The problem is the `PermissionError` exception.

File: conda/core/package_cache_data.py [hunk 1]
```diff
@@ -429,11 +429,11 @@ class UrlsData(object):
     def __init__(self, pkgs_dir):
         self.pkgs_dir = pkgs_dir
         self.urls_txt_path = urls_txt_path = join(pkgs_dir, 'urls.txt')
-        if isfile(urls_txt_path):
+        try:
             with open(urls_txt_path, 'rb') as fh:
                 self._urls_data = [line.strip().decode('utf-8') for line in fh]
                 self._urls_data.reverse()
-        else:
+        except (PermissionError, FileNotFoundError):
```
Comment (mcg1969) [inline conda/core/package_cache_data.py:436]: Need an option for Python 2 here?

**Comment by cjmartian** [comment]
@cla-bot check

**Comment by mcg1969** [comment]
OK, this looks good. @cjmartian, the author has claimed in his other PR that the CLA was signed...

**Comment by cjmartian** [comment]
@cla-bot check

**Comment by cjmartian** [comment]
> OK, this looks good. @cjmartian, the author has claimed in his other PR that the CLA was signed...

I'll go double check @mcg1969

**Comment by cjmartian** [comment]
@bfis Sorry for the trouble, could you please sign the cla document again? Looks like there were some issues on our end.

**Comment by mcg1969** [comment]
Actually @bfis I've found it

**Comment by mcg1969** [comment]
@bfis I now see the problem. It looks like you used your GitHub handle in both the Name and Signature fields, but unfortunately we need the contract to be signed with your full name. A quick perusal of your GitHub activity and Google activity indicates that you prefer anonymity. And while we can certainly respect that in the repository itself, we would need the real name on the signature page (which nobody will see once it is filed away). If you're OK with that, please go ahead and sign the CLA again as @cjmartian suggests. If not we'll need to find another avenue to get these useful fixes in place...

**Comment by bfis** [comment]
FYI: I've signed the CLA again with my full name (on 4/12/21).

**Comment by jezdez** [comment]
@anaconda-issue-bot check

**Comment by travishathaway** [comment]
This pull request has become stale and is therefore being closed. Please re-open if you would like to continue development on this.

**Pull request closed without merging**